### DISCUSSION TITLE
There is no option to purge cache on vercel

### DISCUSSION THREAD

**Discussion by yevyevyev** [vercel/next.js#21109] [Help]
How to manually purge cache? Is there a programmatic access to cache? This is crucial for my app since people can create user pages but then when they delete the data is deleted but next.js stores the cache for 31 day.

**Comment by michrome** [comment]
Hey @tapkain you have to set the [Cache-Control Header](https://vercel.com/docs/edge-network/headers?query=s-maxage#cache-control-header)  to a value you are comfortable with.

There is no way to remove a cache key via the platform API.

**Reply by yevyevyev** [reply]
Hello @michrome thank you. so basically to be more specific in the context of my problem, each time I'm returning the data from getStaticProps I need to check if it was deleted and if it was -> set Cache-Control Header s-maxage to 0?

because right now if the data was not found I return not found obviously but I suppose next.js uses the cached version as a fallback for such case?

**Reply by michrome** [reply]
Are you running `getStaticProps`? If so, you’re going to replace what’s in the cache, because you’re either creating a new deploy (which invalidates the cache for the entire project) or you’re using [Incremental Static Regeneration](https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration) (which invalidates the cache for that page).

**Reply by michrome** [reply]
Did this help @tapkain ?
